src/ifcconvert/ColladaSerializer.cpp

```diff
@@ -19,48 +19,91 @@
 
 #ifdef WITH_OPENCOLLADA
 
+#include "ColladaSerializer.h"
+
+#include <COLLADASWPrimitves.h>
+#include <COLLADASWSource.h>
+#include <COLLADASWScene.h>
+#include <COLLADASWNode.h>
+#include <COLLADASWInstanceGeometry.h>
+#include <COLLADASWBaseInputElement.h>
+#include <COLLADASWAsset.h>
+
 #include <string>
+#include <cmath>
 
-#include "ColladaSerializer.h"
+static void collada_id(std::string &s)
+{
+    IfcUtil::sanitate_material_name(s);
+    IfcUtil::escape_xml(s);
+}
 
-std::string collada_id(const std::string& s) {
-	std::string id;
-	id.reserve(s.size());
-	for (std::string::const_iterator it = s.begin(); it != s.end(); ++it) {
-		const std::string::value_type c = *it;
-		if ((c >= '0' && c <= '9') || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c == '_') || ( c == '-')) {
-			id.push_back(c);
-		}
-	}
-	return id;
+/// @todo Very simple impl. Assumes that input vertices and normals match 1:1.
+static std::vector<real_t> box_project_uvs(const std::vector<real_t> &vertices, const std::vector<real_t> &normals)
```

Comment:
Nice. Ideally this would be moved to `IfcGeom::Representation::Triangulation` and uvs added as member vector. Then it would be trivial to re-use this for obj/blender/max/etc.